chips/cc26x2/src/rtc.rs

```diff
@@ -0,0 +1,195 @@
+//! RTC driver
+
+use core::cell::Cell;
+use kernel::common::registers::{ReadOnly, ReadWrite};
+use kernel::common::StaticRef;
+use kernel::hil::time::{self, Alarm, Frequency, Time};
+
+#[repr(C)]
+struct RtcRegisters {
+    ctl: ReadWrite<u32, Control::Register>,
+
+    // Event flags
+    evflags: ReadWrite<u32, EvFlags::Register>,
+
+    // Integer part
+    sec: ReadWrite<u32>,
+    // Fractional part (1/32kHz parts of a second)
+    subsec: ReadOnly<u32>,
+
+    _subsec_inc: ReadOnly<u32>,
+    channel_ctl: ReadWrite<u32, ChannelControl::Register>,
+    _channel0_cmp: ReadOnly<u32>,
+    channel1_cmp: ReadWrite<u32>,
+    _channel2_cmp: ReadOnly<u32>,
+    _channel2_cmp_inc: ReadOnly<u32>,
+    _channel1_capture: ReadOnly<u32>,
+
+    // A read request to the sync register will not return
+    // until all outstanding writes have properly propagated to the RTC domain
+    sync: ReadOnly<u32>,
+}
+
+register_bitfields![
+    u32,
+    Control [
+        COMB_EV_MASK OFFSET(16) NUMBITS(3) [
+            NoEvent = 0b00,
+            Channel0 = 0b01,
+            Channel1 = 0b10,
+            Channel2 = 0b11
+        ],
+        RESET       OFFSET(7) NUMBITS(1) [],
+        RTC_UPD_EN  OFFSET(1) NUMBITS(1) [],
+        ENABLE      OFFSET(0) NUMBITS(1) []
+    ],
+    EvFlags [
+        CH2 OFFSET(16) NUMBITS(1) [],
+        CH1 OFFSET(8)  NUMBITS(1) [],
+        CH0 OFFSET(0)  NUMBITS(1) []
+    ],
+    ChannelControl [
+        CH2_CONT_EN OFFSET(18)  NUMBITS(1) [],
+        CH2_EN      OFFSET(16)  NUMBITS(1) [],
+        CH1_CAPT_EN OFFSET(9)   NUMBITS(1) [],
+        CH1_EN      OFFSET(8)   NUMBITS(1) [],
+        CH0_EN      OFFSET(0)   NUMBITS(1) []
+    ]
+];
+
+const RTC_BASE: StaticRef<RtcRegisters> =
+    unsafe { StaticRef::new(0x40092000 as *const RtcRegisters) };
+
+pub struct Rtc {
+    registers: StaticRef<RtcRegisters>,
+    callback: Cell<Option<&'static time::Client>>,
+}
+
+pub static mut RTC: Rtc = Rtc::new();
+
+impl Rtc {
+    const fn new() -> Rtc {
+        Rtc {
+            registers: RTC_BASE,
+            callback: Cell::new(None),
+        }
+    }
+
+    pub fn start(&self) {
+        let regs = &*self.registers;
+        regs.ctl.write(Control::ENABLE::SET);
+
+        regs.sync.get();
+    }
+
+    pub fn stop(&self) {
+        let regs = &*self.registers;
+        regs.ctl.write(Control::ENABLE::CLEAR);
+
+        regs.sync.get();
+    }
+
+    pub fn sync(&self) {
```

Comment:
Comment describing what this does (really comments for all of these would be good, but this is the most ambiguous, and not really related to the functionality if the RTC.